candidatepair.go

```diff
@@ -100,3 +107,46 @@ func (a *Agent) sendSTUN(msg *stun.Message, local, remote Candidate) {
 		a.log.Tracef("Failed to send STUN message: %s", err)
 	}
 }
+
+// UpdateRoundTripTime sets the current round time of this pair and
+// accumulates total round trip time and responses received
+func (p *CandidatePair) UpdateRoundTripTime(rtt time.Duration) {
+	p.currentRoundTripTime.Store(&rtt)
+
+	prevTotalRoundTripTime := p.totalRoundTripTime.Load()
+	totalRoundTripTime := rtt
+	if prevTotalRoundTripTime != nil {
+		totalRoundTripTime += *prevTotalRoundTripTime
+	}
+	p.totalRoundTripTime.CompareAndSwap(prevTotalRoundTripTime, &totalRoundTripTime)
```

Comment:
Can this just be Store?

Reply:
The following sequence will get the wrong value
1. goroutine A reads value A
2. Another update happens and goroutine B reads value A
3. goroutine A adds to the total (value A + rtt A)
4. goroutine B adds to the total (value A + rtt B) <-- this is the problematic 
5. Depending on who write first, the final value could be `value A + rtt A` or `value A + rtt B`. Both are wrong. The correct value is `value A + rtt A + rtt B`.

There is no strong reason to use `CompareAndSwap`. `Store` would store both values whatever order it happens in. `CompareAndSwap` will store only the first value.

I think I am doing to change to this to just `int64` and do `Duration.Nanoseconds()` so that I can use `atomic.AddInt64()`.

Will make that change now.